rust/private/rust_analyzer.bzl

```diff
@@ -0,0 +1,205 @@
+# Copyright 2020 Google
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Rust Analyzer Bazel rules.
+
+rust_analyzer will generate a rust-project.json file for the
+given targets. This file can be consumed by rust-analyzer as an alternative
+to Cargo.toml files.
+"""
+
+load("@io_bazel_rules_rust//rust:private/rustc.bzl", "BuildInfo", "CrateInfo")
+load("@io_bazel_rules_rust//rust:private/utils.bzl", "find_toolchain")
+load("//rust/platform:triple_mappings.bzl", "system_to_dylib_ext", "triple_to_system")
+
+# We support only these rule kinds.
+_rust_rules = [
+    "rust_library",
+    "rust_binary",
+]
+
+RustAnalyzerInfo = provider(
+    "RustAnalyzerInfo holds rust crate metadata for targets",
+    fields = {
+        "crate": "CrateInfo",
+        "deps": "List[RustAnalyzerInfo]: direct dependencies",
+        "transitive_deps": "List[RustAnalyzerInfo]: transitive closure of dependencies",
+        "cfgs": "List[String]: features or other compilation --cfg= settings",
+        "env": "Dict{String: String}: Environment variables, used for the `env!` macro",
+        "proc_macro_dylib_path": "File: compiled shared library output of proc-macro rule",
+        "build_info": "BuildInfo: build info for this crate if present",
+    },
+)
+
+def _rust_analyzer_aspect_impl(target, ctx):
+    if CrateInfo not in target:
+        return []
+
+    toolchain = find_toolchain(ctx)
+
+    cfgs = []
+    for feature in ctx.rule.attr.crate_features:
+        cfgs.append("feature=\"" + feature + "\"")
+    for flag in ctx.rule.attr.rustc_flags:
+        # --cfg flags should be passed as well but as an atomic
+        # config, not key/value
+        if flag.startswith("--cfg"):
```

Comment:
Should this not be `--cfg=`?

Reply:
The `=` is optional, and could be a space. Updated to be explicit about both possibilities.

Reply:
Ah cool 👍 

The `[6:]` was throwing me off and seemed like an oversight. Thanks for updating that 🙏